invenio_oaiserver/percolator.py

```diff
@@ -110,61 +113,123 @@ def _new_percolator(spec, search_pattern):
         for index, mapping_path in current_search.mappings.items():
             # Create the percolator doc_type in the existing index for >= ES5
             # TODO: Consider doing this only once in app initialization
-            percolator_doc_type = _get_percolator_doc_type(index)
-            _create_percolator_mapping(
-                index, percolator_doc_type, mapping_path)
-            current_search_client.index(
-                index=_build_percolator_index_name(index),
-                doc_type=percolator_doc_type,
-                id='oaiset-{}'.format(spec),
-                body={'query': query}
-            )
+            try:
+                percolator_doc_type = _get_percolator_doc_type(index)
+                _create_percolator_mapping(
+                    index, percolator_doc_type, mapping_path)
+                current_search_client.index(
+                    index=_build_percolator_index_name(index),
+                    doc_type=percolator_doc_type,
+                    id='oaiset-{}'.format(spec),
+                    body={'query': query}
+                )
+            except Exception as e:
+                # caught on schemas, which do not contain the query field
+                current_app.logger.warning(e)
+                pass
 
 
 def _delete_percolator(spec, search_pattern):
-    """Delete percolator associated with the new oaiset."""
-    if spec:
-        for index in current_search.mappings.keys():
-            # Create the percolator doc_type in the existing index for >= ES5
-            percolator_doc_type = _get_percolator_doc_type(index)
-            _create_percolator_mapping(index, percolator_doc_type)
```

Comment:
After talking with @rekt-hard we have deemed this line unnecesary... any ideas why it was there @slint @lnielsen